validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -99,16 +102,45 @@ def broadcast_batch_by_batch_id_request(self, batch_id):
             batch_request,
             validator_pb2.Message.GOSSIP_BATCH_BY_BATCH_ID_REQUEST)
 
-    def broadcast(self, gossip_message, message_type):
-        # Gossip broadcasts are sent out in two different ways.
-        # 1) To connected identities on our server socket
-        # 2) To outbound connections we have originated
+    def broadcast(self, gossip_message, message_type, exclude=None):
+        """Broadcast gossip messages.
+
+        Broadcast to both connected identities on our server socket and to
+        outboud connections we have originated. If a peer's identifiers are in
+        exclude, do not broadcast message to them.
+
+        Args:
+            gossip_message: The message to be broadcast.
+            message_type: Type of the message.
+            exclude: A list of tuples that contains a peer's information
+                (connection, identifier)
+        """
+        if exclude is None:
+            exclude = []
+
+        identities = copy.copy(self._identities)
+        connections = copy.copy(self._network.connections)
+
+        # Remove any identities that are in the exclude list
         for identity in self._identities:
+            for peer in exclude:
+                if peer[1] == identity:
+                    identities.remove(identity)
+
+        # Remove any connections in the exclude list
+        for connection in self._network.connections:
+            for peer in exclude:
+                if peer[0] == connection.local_id:
+                    connections.remove(connection)
+
+        for identity in identities:
```

Comment:
instead of making copies and modifying list objects, you could probably do set subtraction here in the loop or just put a conditional check in to determine if you send or not